wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/ast2wdlom/AstNodeToExpressionElement.scala

```diff
@@ -14,26 +17,196 @@ import wom.values._
+    "write_json" -> validateOneParamEngineFunction(ReadString, "write_json"),
+    "range" -> validateOneParamEngineFunction(ReadString, "range"),
+    "transpose" -> validateOneParamEngineFunction(ReadString, "transpose"),
+    "length" -> validateOneParamEngineFunction(ReadString, "length"),
+    "prefix" -> validateOneParamEngineFunction(ReadString, "prefix"),
+    "select_first" -> validateOneParamEngineFunction(ReadString, "select_first"),
+    "select_all" -> validateOneParamEngineFunction(ReadString, "select_all"),
+    "defined" -> validateOneParamEngineFunction(ReadString, "defined"),
+    "basename" -> validateOneParamEngineFunction(ReadString, "basename"),
+    "floor" -> validateOneParamEngineFunction(ReadString, "floor"),
+    "ceil" -> validateOneParamEngineFunction(ReadString, "ceil"),
+    "round" -> validateOneParamEngineFunction(ReadString, "round"),
+
+    // 1- or 2-param functions:
+    "size" -> validateSizeEngineFunction,
+
+    // 2-param functions:
+    "zip" -> validateTwoParamEngineFunction(Zip, "zip"),
+    "cross" -> validateTwoParamEngineFunction(Cross, "cross"),
+
+    // 3-param functions:
+    "sub" -> validateThreeParamEngineFunction(Sub, "sub"),
+  )
+
+  private def validateNoParamEngineFunction(element: ExpressionElement, functionName: String)
+                                           (params: Vector[ExpressionElement]): ErrorOr[ExpressionElement] =
+    if (params.isEmpty) {
+      element.validNel
+    } else {
+      s"Function $functionName expects 0 arguments but got ${params.size}".invalidNel
+    }
+
+  private def validateOneParamEngineFunction(elementMaker: ExpressionElement => ExpressionElement, functionName: String)
+                                            (params: Vector[ExpressionElement]): ErrorOr[ExpressionElement] =
+    if (params.size == 1) {
+      elementMaker.apply(params.head).validNel
+    } else {
+      s"Function $functionName expects exactly 1 argument but got ${params.size}".invalidNel
+    }
+
+  private def validateSizeEngineFunction(params: Vector[ExpressionElement]): ErrorOr[ExpressionElement] =
+    if (params.size == 1) {
+      Size(params.head, None).validNel
+    } else if (params.size == 2) {
+      Size(params.head, Option(params(1))).validNel
+    } else {
+      s"Function size expects 1 or 2 arguments but got ${params.size}".invalidNel
+    }
+
+  private def validateTwoParamEngineFunction(elementMaker: (ExpressionElement, ExpressionElement) => ExpressionElement, functionName: String)
+                                            (params: Vector[ExpressionElement]): ErrorOr[ExpressionElement] =
+    if (params.size == 2) {
+      elementMaker.apply(params.head, params(1)).validNel
+    } else {
+      s"Function $functionName expects exactly 2 arguments but got ${params.size}".invalidNel
+    }
+
+  private def validateThreeParamEngineFunction(elementMaker: (ExpressionElement, ExpressionElement, ExpressionElement) => ExpressionElement, functionName: String)
```

Comment:
Is three the maximum number of parameters taken by an engine function?

Reply:
I guess in theory they could take any number but right now, there are no engine functions which take more than 3